modules/imgproc/src/moments.cpp

```diff
@@ -265,6 +265,65 @@ struct MomentsInTile_SIMD<uchar, int, int>
     int CV_DECL_ALIGNED(16) buf[4];
 };
 
+template <>
+struct MomentsInTile_SIMD<ushort, int, int64>
+{
+    MomentsInTile_SIMD()
+    {
+        // nothing
+    }
+
+    int operator() (const ushort * ptr, int len, int & x0, int & x1, int & x2, int64 & x3)
+    {
+        int x = 0;
+
+        {
+            v_int32x4 v_delta = v_setall_s32(4), v_ix0 = v_int32x4(0, 1, 2, 3);
+            v_uint32x4 z = v_setzero_u32(), v_x0 = z, v_x1 = z, v_x2 = z;
+            v_uint64x2 v_x3 = v_reinterpret_as_u64(z);
+
+            for( ; x <= len - 4; x += 4 )
+            {
+                v_int32x4 v_src = v_reinterpret_as_s32(v_load_expand(ptr + x));
+
+                v_x0 += v_reinterpret_as_u32(v_src);
+                v_x1 += v_reinterpret_as_u32(v_src * v_ix0);
+
+                v_int32x4 v_ix1 = v_ix0 * v_ix0;
+                v_x2 += v_reinterpret_as_u32(v_src * v_ix1);
+
+                v_ix1 = v_ix0 * v_ix1;
+                v_src = v_src * v_ix1;
+                v_uint64x2 v_lo, v_hi;
+                v_expand(v_reinterpret_as_u32(v_src), v_lo, v_hi);
+                v_x3 += v_lo + v_hi;
+
+                v_ix0 += v_delta;
+            }
+
+            v_uint32x4 v_x01_lo, v_x01_hi, v_x22_lo, v_x22_hi;
+            v_zip(v_x0, v_x1, v_x01_lo, v_x01_hi);
+            v_zip(v_x2, v_x2, v_x22_lo, v_x22_hi);
+            v_x01_lo += v_x01_hi;
+            v_x22_lo += v_x22_hi;
+            v_float64x2 v_x0122_lo, v_x0122_hi;
+            v_recombine(v_reinterpret_as_f64(v_x01_lo), v_reinterpret_as_f64(v_x22_lo), v_x0122_lo, v_x0122_hi);
+            v_store(buf64, v_reinterpret_as_s64(v_x3));
+            v_store(buf, v_reinterpret_as_s32(v_x0122_lo) + v_reinterpret_as_s32(v_x0122_hi));
+
+            x0 = buf[0];
+            x1 = buf[1];
+            x2 = buf[2];
+            x3 = buf64[0] + buf64[1];
```

Comment:
Perhaps we don't need to force 64F here (at least in such complex form).

This is post-processing, so performance here is not critical.

Could you try this approach:
```
x0 = v_reduce_sum(v_x0);
x1 = v_reduce_sum(v_x1);
x2 = v_reduce_sum(v_x2);
int64 CV_DECL_ALIGNED(16) buf64[2];  // avoid declarations as class fields
v_store_aligned(buf64, v_reinterpret_as_s64(v_x3));
x3 = buf64[0] + buf64[1];
```